src/hil/src/spi.rs

```diff
@@ -108,7 +112,27 @@ pub trait SpiMaster {
     // is any of the read/read_write calls. These functions
     // allow an application to manually control when the
     // CS line is high or low, such that it can issue multi-byte
-    // requests with single byte oprations.
+    // requests with single byte operations.
     fn hold_low(&self);
     fn release_low(&self);
 }
+
+/// SPIMasterDevice provides a chip-specific interface to the SPI Master
+/// hardware. The interface wraps the chip select line so that chip drivers
+/// cannot communicate with different SPI devices.
+pub trait SPIMasterDevice {
```

Comment:
I'm not sure I've really thought about what I'm saying. I think that if we have something called `SPIDevice` it has to work as both a master and slave somehow (whenever that code exists). Otherwise I think it will be confusing to need to have both `SPISlave` and `SPIDevice` objects.

I'm also pro punting this. We are going to need `I2CSlave` soon, so it might be better to pick a naming scheme when that is more fully developed.